ui/src/components/view/InfoCard.vue

```diff
@@ -35,7 +35,7 @@
                       <resource-icon :image="resourceIcon" size="4x" style="margin-right: 5px"/>
                     </span>
                     <span v-else>
-                      <os-logo v-if="resource.ostypeid || resource.ostypename || ['guestoscategory'].includes($route.path.split('/')[1])" :osId="resource.ostypeid" :osName="resource.ostypename || resource.name" size="3x" @update-osname="setResourceOsType"/>
+                      <os-logo v-if="resource.ostypeid || resource.ostypename || ['guestoscategory'].includes($route.path.split('/')[1])" :osId="resource.ostypeid" :osName="resource.ostypename || resource.osdisplayname || resource.name" size="3x" />
```

Comment:
The core issue is passing resource.name to osName,
while os-logo only does lookup by osId when osName is empty.

Removal of the @update-osname also makes it incorrect,
as it's now showing OS instead of OS type, e.g. CentOS 9 instead of CentOS.